src/main/java/com/angelbroking/smartapi/http/SmartAPIResponseHandler.java

```diff
@@ -87,10 +99,64 @@ private SmartAPIException dealWithException(JSONObject jsonObject, String code)
 		case "AB1001":
 		case "AB1011":
 			return new PermissionException(jsonObject.getString("message"), code);
-
+            case "AG8004":
+                return new ApiKeyException(APIKEY_EXCEPTION_MESSAGE, code);
 		default:
 			return new SmartAPIException(jsonObject.getString("data not found"));
 		}
 	}
 
+    public String handler(Response response, String body) throws SmartAPIException, JSONException, IOException {
+        if (response.code() == 200) {
+            return handleResponse(response,body);
+        } else if (response.code() == 400){
+            log.error("Invalid exchange provided. Please provide a valid exchange");
+            return "Invalid exchange provided. Please provide a valid exchange";
+        }else {
+            log.error("Response or response body is null.");
+            throw new IllegalArgumentException("Response or response body is null.");
+        }
+    }
+
+    private String handleResponse(Response response, String body) throws SmartAPIException, IOException {
+        try {
+            JSONObject responseBodyJson = new JSONObject(body);
+            String errorCode = responseBodyJson.optString("errorCode");
+            if(errorCode == null || errorCode.length() == 0) {

```

Comment:
if(errorCode == null || errorCode.length() == 0) { 

What does this line mean? Value of errorCode?

Reply:
for this condition, we checked if the errorCode is null or since its in String we also check its size , if its equal to 0,

that means if this condition is true we have no error while sending the response and we can proceed with the given response

Reply:
for this condition, we checked if the errorCode is null or since its in String we also check its size , if its equal to 0,

that means if this condition is true we have no error while sending the response and we can proceed with the given response

Reply:
have changed the code to as requested

   if(responseBodyJson.getBoolean("status")) {